src/controller/src/clusters.rs

```diff
@@ -855,7 +765,7 @@ async fn try_remove_past_generation_replicas(
                 replica_id = %name.replica_id,
                 "removing past generation replica",
             );
-            orchestrator.drop_service(&service).await?;
+            orchestrator.drop_service(&service)?;
         }
     }
 
```

Comment:
The code in this file got so much cleaner. 👌🏽 Really makes me feel like we got the abstraction right. Probably shouldn't have embarked on parallelizing the replica creation requests, and instead invested in the background task from the get go. Oh well.